core/src/main/scala/slamdata/interpreters/PullRequestsInterpreter.scala

```diff
@@ -0,0 +1,155 @@
+    client.patch[PullRequestUpdate, PullRequestDraft](accessToken, s"repos/$owner/$repo/pulls/$number", draftHeaders, fields)
+  }
+
+  def markReadyForReview(owner: String,
+                         repo: String,
+                         id: String,
+                         headers: Map[String, String]): F[GHResponse[Boolean]] = {
+    val draftHeaders: Map[String, String] = headers + draftHeader
+    val data = new GithubQuery(MarkForReview, MarkForReview.Variables(id))
+    client
+      .post[GithubQuery[MarkForReview.type], GithubResponse[MarkForReview.type]](accessToken, "graphql", draftHeaders, data)
+      .map(_.map(r => r.copy(result = r.result.data.markPullRequestReadyForReview.exists(_.pullRequest.exists(!_.isDraft)))))
+  }
+
+  def getPullRequest(owner: String, repo: String, number: Int, headers: Map[String, String]): F[GHResponse[PullRequest]] =
+    impl.getPullRequest(owner, repo, number, headers)
+
+  def listPullRequests(owner: String,
+                       repo: String,
+                       filters: List[PRFilter],
+                       pagination: Option[Pagination],
+                       headers: Map[String, String]): F[GHResponse[List[PullRequest]]] =
+    impl.listPullRequests(owner, repo, filters, pagination, headers)
+
+  def listFiles(owner: String,
+                repo: String,
+                number: Int,
+                pagination: Option[Pagination],
+                headers: Map[String, String]): F[GHResponse[List[PullRequestFile]]] =
+    impl.listFiles(owner, repo, number, pagination, headers)
+
+  def createPullRequest(owner: String,
+                        repo: String,
+                        newPullRequest: NewPullRequest,
+                        head: String,
+                        base: String,
+                        maintainerCanModify: Option[Boolean],
+                        headers: Map[String, String]): F[GHResponse[PullRequest]] =
+    impl.createPullRequest(owner, repo, newPullRequest, head, base, maintainerCanModify, headers)
+
+  def listReviews(owner: String,
+                  repo: String,
+                  pullRequest: Int,
+                  pagination: Option[Pagination],
+                  headers: Map[String, String]): F[GHResponse[List[PullRequestReview]]] =
+    impl.listReviews(owner, repo, pullRequest, pagination, headers)
+
+  def getReview(owner: String,
+                repo: String,
+                pullRequest: Int,
+                review: Int,
+                headers: Map[String, String]): F[GHResponse[PullRequestReview]] =
+    impl.getReview(owner, repo, pullRequest, review, headers)
+
+}
+
+object PullRequestsInterpreter {
+  import io.circe._
+  import io.circe.generic.auto._
+  import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}
+  import io.circe.syntax._
+  import github4s.Decoders._
+
+  /** Representation of the json data github graphql queries expect. */
+  case class GithubQuery[A <: GraphQLQuery](query: A, variables: A#Variables)
```

Comment:
First off, `final`. Second, this is a surprisingly tricky type particularly taken into context with the implicits in the companion object. Is it actually strictly necessary?